### PR TITLE
Manually dedupe entries when listing files in s3 

### PR DESCRIPTION
**Pull request by colinmarc**
It's not clear why this is necessary, but we've been seeing duplicate
entries fairly often.

We should wait and see if #43 magically fixed this, though.

### CODE REVIEW

**Comment by colinmarc** [comment]
Nope, #43 did not magically fix this.

File: backend/s3_backend.go [hunk 1]
```diff
@@ -142,8 +144,13 @@ func (s *S3Backend) ListFiles(db, version string) ([]string, error) {
 		return nil, s.s3error(err)
 	}
 
-	sort.Strings(res)
-	return res, nil
+	sorted := make([]string, 0, len(res))
```
Comment (forklady42) [inline backend/s3_backend.go]: Kinda amused that this is called sorted even before it's sorted, but it's being created in order to be sorted, so fine.

**Comment by forklady42** [comment]
👍

File: backend/s3_backend.go (hunk 1 above)
Comment (colinmarc) [inline backend/s3_backend.go]: I actually was kinda stuck on what to name it.

**Pull request merged**